{
  Closure<Integer> work=createMock(INT_CLOSURE_CLZ);
  work.execute(1);
  RuntimeException runtimeException=new IllegalArgumentException();
  EasyMock.expectLastCall().andThrow(runtimeException);
  control.replay();
  Function<Integer,Void> workFunction=Closures.asFunction(work);
  try {
    workFunction.apply(1);
  }
 catch (  RuntimeException e) {
    assertSame(runtimeException,e);
  }
}
